src/qibolab/result.py

```diff
@@ -1,10 +1,164 @@
+
+    def __add__(self, data):
+        new_res = super().__add__(data)
+        new_res.std = np.append(self.std, data.std)
+        return new_res
+
+
+class RawWaveformResults(IntegratedResults):
+    """
+    Data structure to deal with the output of
+    :func:`qibolab.platforms.abstract.AbstractPlatform.execute_pulse_sequence`
+    :func:`qibolab.platforms.abstract.AbstractPlatform.sweep`
+
+    Associated with AcquisitionType.RAW and AveragingMode.SINGLESHOT
+    may also be used to store the integration weights ?
+    """
+
+
+class AveragedRawWaveformResults(AveragedIntegratedResults):
+    """
+    Data structure to deal with the output of
+    :func:`qibolab.platforms.abstract.AbstractPlatform.execute_pulse_sequence`
+    :func:`qibolab.platforms.abstract.AbstractPlatform.sweep`
+
+    Associated with AcquisitionType.RAW and AveragingMode.CYCLIC
+    or the averages of ``RawWaveformResults``
+    """
+
+
+class StateResults:
+    """
+    Data structure to deal with the output of
+    :func:`qibolab.platforms.abstract.AbstractPlatform.execute_pulse_sequence`
+    :func:`qibolab.platforms.abstract.AbstractPlatform.sweep`
+
+    Associated with AcquisitionType.DISCRIMINATION and AveragingMode.SINGLESHOT
+    """
+
+    def __init__(self, data: np.ndarray):
+        self.states: npt.NDArray[np.uint32] = data
+
+    def __add__(self, data):
+        return self.__class__(np.append(self.states, data.states))
+
+    @lru_cache
+    def probability(self, state=0):
+        """Returns the statistical frequency of the specified state (0 or 1)."""
+        return abs(1 - state - np.mean(self.states, axis=0))
+
+    @property
+    def serialize(self):
+        """Serialize as a dictionary."""
+        serialized_dict = {
+            "state_0": self.probability(0).flatten(),
+        }
+        return serialized_dict
+
+    @property
+    def average(self):
+        """Perform states average"""
+        average = self.probability(1)
+        std = np.std(self.states, axis=0, ddof=1) / np.sqrt(self.states.shape[0])
+        return AveragedStateResults(average, std=std)
+
+
+# FIXME: Here I take the states from StateResult that are typed to be ints but those are not what would you do ?
+class AveragedStateResults(StateResults):
+    """
+    Data structure to deal with the output of
+    :func:`qibolab.platforms.abstract.AbstractPlatform.execute_pulse_sequence`
+    :func:`qibolab.platforms.abstract.AbstractPlatform.sweep`
+
+    Associated with AcquisitionType.DISCRIMINATION and AveragingMode.CYCLIC
+    or the averages of ``StateResults``
+    """
+
+    def __init__(self, states: np.ndarray, std: np.ndarray = np.array([])):
+        super().__init__(states)
+        self.std: Optional[npt.NDArray[np.float64]] = std
+
+    def __add__(self, data):
+        new_res = super().__add__(data)
+        new_res.std = np.append(self.std, data.std)
+        return new_res
+
+
 ExecRes = np.dtype([("i", np.float64), ("q", np.float64)])
```

Comment:
@Jacfomg thank you for all the updates, it looks good to me too. I will merge this to my branch and check with QM.

When I was merging I noticed that you still keep the old dtype and dataclasses here. Do you need them? If no we should remove everything under this line.

Reply:
I don't need them, but as qblox is not updated yet they would still be able to do things and update it when they are able.

Reply:
Alvaro wants to introduce this changes on his sweeper branch so he told it's ok to merge and qblox would only work from their branch until they implement all the changes they want